### DISCUSSION TITLE
Evaluate using Profile-Guided Optimization (PGO) for Prisma

### DISCUSSION THREAD

**Discussion by zamazan4ik** [prisma/prisma#26114] [Q&A]
Hi!

Recently I did many Profile-Guided Optimization (PGO) benchmarks on multiple projects (including many databases like PostgreSQL, ClickHouse, Redis, MongoDB, etc.) - the results are available [here](https://github.com/zamazan4ik/awesome-pgo). According to my tests, PGO helps with optimizing performance for many applications in different software domains. I think it can be interesting trying to apply PGO to Prisma libraries and measure the performance improvement from PGO.

I can suggest the following things to do:

* Evaluate PGO on Prisma.
* If PGO helps to achieve better performance - add a note to Prisma's documentation about that. In this case, users and maintainers will be aware of another optimization opportunity for Prisma.
* Provide PGO integration into the build scripts. It can help users and maintainers easily apply PGO for their workloads.
* Optimize prebuilt binaries with PGO if you will be able to figure out a "typical" workload.

PGO can be enabled for both standalone binaries and libraries. For starting with PGO for Rust projects I can recommend using [cargo-pgo](https://github.com/Kobzol/cargo-pgo) tool. However, you also can use PGO-related compiler [flags](https://doc.rust-lang.org/rustc/profile-guided-optimization.html) directly.

P.S. Please do not treat the issue as a bug or smth like that - I just created it since Discussions are disabled for the repository. It's just an idea how Prisma's performance can be improved.

**Comment by jkomyno** [comment]
Hello, I'm moving this issue to prisma/prisma (our main repository, which we monitor for issues much more often than this one).
This way, I can convert this to a discussion there.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
